### PR TITLE
drop nestandard1.1 in favour of 2.0, and update uwp and net versions

### PR DESCRIPTION
**Pull request by olevett**
**What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)**

Somewhat related to https://github.com/reactiveui/ReactiveUI/pull/1579 and also general housekeeping/trying to minimise the targets we're having to support.

**What is the current behavior? (You can also link to an open issue here)**

**What is the new behavior (if this is a feature change)?**

**What might this PR break?**

**Please check if the PR fulfills these requirements**
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Other information**:

FYI @onovotny

### CODE REVIEW

File: src/Directory.build.targets [hunk 1]
```diff
@@ -1,14 +1,11 @@
...
-  </PropertyGroup>
-  <PropertyGroup Condition="'$(TargetFramework)' == 'net45'">
+  <PropertyGroup Condition="'$(TargetFramework)' == 'net461'">
     <DefineConstants>$(DefineConstants);NET_45;XAML</DefineConstants>
   </PropertyGroup>
   <PropertyGroup Condition="'$(TargetFramework)' == 'uap10.0'">
     <DefineConstants>$(DefineConstants);NETFX_CORE;XAML;WINDOWS_UWP</DefineConstants>
-    <TargetPlatformVersion>10.0.14393.0</TargetPlatformVersion>
-    <TargetPlatformMinVersion>10.0.10586.0</TargetPlatformMinVersion>
+    <TargetPlatformVersion>10.0.16299.0</TargetPlatformVersion>
```
Comment (onovotny) [inline src/Directory.build.targets]: This isn't enough here. You don't need to set TPV/TPMV, instead the TFM everywhere should change to be `uap10.0.16299`.

That sets the TPMV based on that automatically.

File: src/Splat/Splat.csproj [hunk 2]
```diff
@@ -1,6 +1,6 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
-    <TargetFrameworks>netstandard1.1;net45;uap10.0;MonoAndroid70;Xamarin.iOS10;Xamarin.Mac20;netstandard2.0</TargetFrameworks>
+    <TargetFrameworks>net461;uap10.0;MonoAndroid70;Xamarin.iOS10;Xamarin.Mac20;netstandard2.0</TargetFrameworks>
```
Comment (onovotny) [inline src/Splat/Splat.csproj]: TFM needs to go to `uap10.0.16299` everywhere.

**Comment by ghuntley** [comment]
Not sure we can do this. Won't it up the min version of ReactiveUI and mean net45x won't be targetable?

**Comment by onovotny** [comment]
Yes...but that's my point. Those are all very old and unsupported. Why do we need to keep supporting those?

**Comment by glennawatson** [comment]
What advantage do you get our of supporting the old .NET framework versions? You can upgrade to the 4.6s in Windows 7 and beyond, and even the .NET 4.7 changes? And other platforms would be targeted by the .NET standard wouldn't they?

**Comment by ghuntley** [comment]
I am being extremely conservative about rev bumps. Is it too soon?

The default version for a brand new console application is now 461 which is supported by netstandard20 which is the metric I use to decide when to ditch an older platform. I'm happy to ditch < 461.

I'm good going for netstandard20 as min but I had a look at the support matrix a couple moments ago and have some new concerns about Xamarin.

https://docs.microsoft.com/en-us/dotnet/standard/net-standard

image

Wouldn't netstandard16 be a better fit? Are we being too agressive?

**Comment by ghuntley** [comment]
The current version of Android in VS2017 8.2.0 and iOS is 11.9. Let's do this. Let's go all in with netstandard20 with splat and reactiveui.

**Comment by olevett** [comment]
We're currently targeting monoandroid70, should we bump this to 8, given the netstandard, or does this at least partly cover off your concern?

**Comment by ghuntley** [comment]
We should bump to MonoAndroid80. That was an oversight on my part. No need to cook new release for it though.

**Pull request merged**